### PR TITLE
Document how to reauthenticate end-user

### LINKED ISSUE #44: Document reauthentication
Doc page: https://docs.authgear.com/integrate/reauthentication

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #44

### CODE REVIEW

**Comment by louischan-oursky** [comment]
https://app.gitbook.com/@louischan/s/louischan-authgear/

**Review by carmenlau — COMMENTED** [review]
The doc overall look good! But seems we only have react native example. First, I think we need to indicate the code is for react native. And we also need to add example for web as well.

File: integrate/reauthentication.md [hunk 1]
```diff
@@ -6,7 +6,304 @@ description: >-
...
-{% hint style="success" %}
-TODO: Explain how re-auth works and provide SDK samples
-{% endhint %}
+Reauthentication in Authgear is built on top of the [OIDC ID token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken). The ID token is a JWT.
+
+Your sensitive operation server endpoint **MUST** require the ID token. When you receive the ID token, you **MUST** verify the signature of it. If the signature is valid, you can trust the claims inside the ID token.
+
+The `auth_time` claim in the ID token tells **when** was the end-user last authenticated. You should check the `auth_time` claim to see if the end-user was authenticated recently enough.
+
+The `https://authgear.com/claims/user/can_reauthenticate` claim in the ID token tells whether the end-user can be reauthenticated. If the value of this claim is `false`, then depending on your business needs, you can either allow the end-user to proceed, or forbid the end-user to perform sensitive operations. The flows are illustrated by the following diagrams.
```
Comment (fungc-io) [inline integrate/reauthentication.md:15]: Does this mean there will be a Claim called `can_reauthenticate` in the ID token?
If so, perhaps change to
> The `can_reauthenticate` claim in the ID token...

Comment (fungc-io) [reply integrate/reauthentication.md:15]: Answer: Custom claims do look like a url

**Pull request closed without merging**